src/rpc/server.h

```diff
@@ -198,7 +198,7 @@ extern UniValue estimatepriority(const UniValue& params, bool fHelp);
 extern UniValue estimatesmartfee(const UniValue& params, bool fHelp);
 extern UniValue estimatesmartpriority(const UniValue& params, bool fHelp);
 
-extern UniValue getnewaddress(const UniValue& params, bool fHelp); // in rpcwallet.cpp
+extern UniValue getnewaddress(const UniValue& params, bool fHelp); // in rpc/wallet.cpp
```

Comment:
#7307 would solve that.
But I agree, changing to `wallet/rpcwallet.cpp` is probably fine for this PR.